### PR TITLE
User Login connected to companions table

### PR DESCRIPTION
**Pull request by andrewMacmurray**
The login is now connected to a db function (rather than an environment variable)

Have added Boom errors, is this a reasonable way to go about it?

#107

### CODE REVIEW

**Comment by andrewMacmurray** [comment]
phew that one was a bit hairy! Circle was doing something very odd with the tests (had to add two which effectively do the same thing but would break when there was only one of each)

File: lib/hapiLoginConfig.js [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
+    const hashedPassword = rows[0].password;
+    bcrypt.compare(password, hashedPassword, (err, isValid) => {
+      if (err) return reply(serverError);
+      if (!isValid) return reply(wrongPassword);
+
       db.getAllData(email, (err, data) => {
-        if (err) throw err;
-        reply(data);
+        if (err) return reply(getDataError);
+        return reply(data);
```
Comment (rjmk) [inline lib/hapiLoginConfig.js]: Note you don't *have* to return this, because there's no more function to execute. It's fine to do so, however

**Comment by rjmk** [comment]
Looks great. :heart: the Boom

**Comment by andrewMacmurray** [comment]
thanks @rjmk ! Will use Boom elsewhere too 😄

**Pull request merged**